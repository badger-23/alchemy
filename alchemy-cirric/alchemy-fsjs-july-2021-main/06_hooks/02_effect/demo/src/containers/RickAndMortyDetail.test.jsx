import React from 'react';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter } from 'react-router';
import detailData from '../fixtures/detail.json';
import RickAndMortyDetail from './RickAndMortyDetail';

const server = setupServer(
  rest.get('https://rickandmortyapi.com/api/character/:id', (req, res, ctx) => {
    return res(ctx.json(detailData));
  })
);

describe('RickAndMortyDetail', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('display a single character', async () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/1']}>
        <RickAndMortyDetail />
      </MemoryRouter>
    );

    screen.getByText('Loading character...');

    await screen.findByText('Rick Sanchez', { exact: false });
    expect(container).toMatchSnapshot();
  });
});
