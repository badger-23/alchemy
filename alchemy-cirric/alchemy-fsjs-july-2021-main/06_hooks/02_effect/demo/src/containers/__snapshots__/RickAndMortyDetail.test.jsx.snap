// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RickAndMortyDetail display a single character 1`] = `
<div>
  <main>
    <a
      href="/"
    >
      Go home
    </a>
    <figure>
      <img
        alt="Rick Sanchez"
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <figcaption>
        Rick Sanchez
         - 
        Human
         - 
        Alive
      </figcaption>
    </figure>
  </main>
</div>
`;
